portal/src/graphql/adminapi/UserDetailsAccountSecurity.tsx

```diff
@@ -350,15 +350,15 @@ const RemoveConfirmationDialog: React.FC<RemoveConfirmationDialogProps> = functi
     <Dialog
       hidden={!visible}
       dialogContentProps={removeConfirmDialogContentProps}
-      onDismiss={onDismiss}
+      onDismiss={deletingAuthenticator ? undefined : onDismiss}
```

Comment:
I think we should also set `modalProps` to make the dialog blocking. https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog

Instead of conditionally passing `onDismiss`, can we check `deletingAuthenticator` in `onDismiss` to determine whether the dialog can be dismissed

Can we make sure all usage of Dialog follow this new convention?

Reply:
> I think we should also set modalProps to make the dialog blocking.

Then clicking background cannot dismiss the dialog. Only block when loading?

> Instead of conditionally passing onDismiss, can we check deletingAuthenticator in onDismiss to determine whether the dialog can be dismissed

If we do so, we do not need to set modalProps?

Reply:
> Then clicking background cannot dismiss the dialog. Only block when loading?

Yes, only block the dialog from dismissal while loading.

> If we do so, we do not need to set modalProps?

Technically you are correct, but let's do both to make our intention clear.